task07-consume-real-data/index.js

```diff
@@ -0,0 +1,143 @@
+        return json;
+    }
+    catch(error) {
+        console.error(`Could not get data: ${error}`);
+    }
+}
+function compareStrings(a, b) {
+    return (a < b) ? -1 : (a > b) ? 1 : 0;
+}
+
+const paintData = ()=>{
+    const json = readRealData();
+    json.then((result)=>{
+        let sortedIslands = result.islands.sort(function(a, b) {
+            return compareStrings(a.name, b.name);
+        })
+        let sortedCharacters = result.characters.sort(function(a, b) {
+            return compareStrings(a.name, b.name);
+        })
+        let sortedMysticObjects = result.mysticObjects.sort(function(a, b) {
+            return compareStrings(a.name, b.name);
+        })
+        populateCharacters(sortedCharacters);
+        populateIslands(sortedIslands);
+        populateMysticObjects(sortedMysticObjects);
+    })
+}
+
+
+
+const populateCharacters = (charactersList)=>{
+    let  charactersGrid = document.getElementById('characters-grid');
+    let charactersListSlice = charactersList.slice(0,4);
+    charactersListSlice.forEach((element,index) => {
+        let html = charactersHTML(element);
+        charactersGrid.innerHTML = charactersGrid.innerHTML + html;
+    });
+}
+const charactersHTML = (character) => {
+    const htmlStructure = `<div class="characters-container_grid_item">
+                                <img src="images/${character.img}" alt="" />
+                                <p class="characters-container_grid_item_characterName">
+                                    ${character.name}
+                                </p>
+                            </div>`
+    return htmlStructure;
+}
+
+
+const populateIslands = (islandsList)=>{
+    let  islandGrid = document.getElementById('island-grid');
+    let islandListSlice = islandsList.slice(0,3);
+    islandListSlice.forEach((element,index) => {
+        let html = islandsHTML(element);
+        islandGrid.innerHTML = islandGrid.innerHTML + html;
+    });
+}
+
+
+const islandsHTML = (island) => {
+        const htmlStructure = `<div class="island-container_grid_item">
+                                    <img src="images/${island.img}" alt="" />
+                                    <div class="island-container_grid_item_descripcion">
+                                    <b><span>${island.name}</span></b>
+                                    <div class="island-container_grid_item_descripcion_region">
+                                        <img
+                                        style="width: 10px; height: 10px"
+                                        src="images/loc.svg"
+                                        alt=""
+                                        />
+                                        <span>${island.Location}</span>
+                                    </div>
+                                    </div>
+                                </div>`
+    return htmlStructure;
+}
+
+
+const populateMysticObjects = (mysticObjectsList)=>{
```

Comment:
All `populate` functions have pretty much the same logic, make it a single function and pass whats different as argument

Reply:
Done!!